kernel/fatfs.c

```diff
@@ -133,10 +134,46 @@ STATIC void init_direntry(struct dirent *dentry, unsigned attrib,
 int dos_open(char *path, unsigned flags, unsigned attrib, int fd)
 {
   REG f_node_ptr fnp = sft_to_fnode(fd);
-  int status = find_fname(path, D_ALL | attrib, fnp);
+  int status;
+
+  /* Special handling for volume name (they can coexist with files/directories)
+   */
+  if (((attrib & (D_DIR | D_VOLID | D_SYSTEM | D_HIDDEN | D_RDONLY)) == D_VOLID) && (flags & O_CREAT))
```

Comment:
What if some other attrs are set too?
Don't you need to check only D_VOLID?

Reply:
> D_VOLID + D_SYSTEM + D_HIDDEN + D_RDONLY (I believe this is an LFN entry)

Could you please clarify, LFN entry
means all of these set together? In
that case the check is wrong as it
looks for either one.
Is D_VOLID|D_RDONLY forbidden?
I would guess VOL is quite readonly. :)

Reply:
https://en.wikipedia.org/wiki/Design_of_the_FAT_file_system#VFAT_long_file_names this seems to suggest that LFN entries have attributes of 0x0f

Reply:
In that case I think you need something
like `(attrib & D_VOLID) && (attrib & 0x0f) != 0x0f`.

Also what do we know about this PR #208 
with DR-DOSish delete logic? You indicated
that you won't be able to write a new tests
for it any soon, but does it pass the current
ones? Do we have a basic tests like delete
file on FAT and make sure its no longer there?
I guess if so then I'll just merge it.

Reply:
> What if some other attrs are set too?
Don't you need to check only D_VOLID?

```c
if (((attrib & (D_DIR | D_VOLID | D_SYSTEM | D_HIDDEN | D_RDONLY)) == D_VOLID) && (flags & O_CREAT))
```

I think that's what this does  (it ands all the relevant bits in attrib then sees if the result is equal to VOLID) , but I admit it's not very readable. I wanted it to be as restrictive as possible. Happy to change however it to whatever you fancy?

Reply:
Why do you think its equal?
Your check doesn't permit D_VOLID|D_RDONLY
and &0xf!=0xf permits such but doesn't permit LFN.

Reply:
I'm looking at this again. I now have 
```c
  if ((attrib & D_VOLID) &&
      ((attrib & D_DIR) != D_DIR) && ((attrib & D_LFN) != D_LFN) &&
      (flags & O_CREAT) && (flags & O_FCB))
  {
```